src/libstd/sys/unix/fs.rs

```diff
@@ -511,10 +511,19 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
 }
 
 pub fn remove_dir_all(path: &Path) -> io::Result<()> {
+    let filetype = try!(lstat(path)).file_type();
+    if filetype.is_symlink() {
+        unlink(path)
+    } else {
+        remove_dir_all_recursive(path)
+    }
+}
```

Comment:
This logic looks pretty similar between here and windows, perhaps it could be lifted to `std::fs` and leave the recursive bits to the platform-specific modules?